graphbolt/src/csc_sampling_graph.cc

```diff
@@ -131,16 +131,144 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
           : torch::nullopt);
 }
 
+NumPickFn GetNumPickFnWithoutEtype(
+    int64_t fanout, bool replace,
+    const torch::optional<torch::Tensor>& probs_or_mask) {
+  if (!probs_or_mask.has_value()) {
+    return [fanout, replace](int64_t offset, int64_t num_neighbors) {
+      int64_t max_count = fanout == -1 ? num_neighbors : fanout;
+      if (replace) {
+        return num_neighbors == 0 ? 0 : max_count;
+      } else {
+        return std::min(num_neighbors, max_count);
+      }
+    };
+  } else {
+    return [fanout, replace, probs_or_mask](
+               int64_t offset, int64_t num_neighbors) {
+      int64_t max_count = fanout == -1 ? num_neighbors : fanout;
+      int64_t cur_count = 0;
+      AT_DISPATCH_ALL_TYPES(
```

Comment:
Why don't use torch operation to get the result?
There should be some optimization inside like vectorlization instruction.